src/reactive-membrane.ts

```diff
@@ -53,12 +53,23 @@ function defaultValueIsObservable(value: any): boolean {
     if (value == null) {
         return false;
     }
-    if (isArray(value)) {
-        return true;
-    }
 
-    const proto = getPrototypeOf(value);
-    return (proto === ObjectDotPrototype || proto === null || getPrototypeOf(proto) === null);
+    switch (typeof value) {
```

Comment:
have you tried the inverse of this? `if (typeof value !== 'object')` or maybe `if (typeof value === 'object')` whatever is faster. This is a single condition, we don't need all the other stuff...